Support for dependency locking, do not use absolute paths

### What happened?

Here https://github.com/JetBrains/intellij-platform-gradle-plugin/blob/33af9bdd1d0d8b996b911781ab23013dd14997f4/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt#L1097

The plugin does
```kotlin
private val Path.hash
    get() = (absolutePathString().hashCode().absoluteValue % 1000).toString()
```
which then is used to create dependencies in Gradle, also in transforms.

```
Property
    Path.hash
Usages in All Places  (7 usages found)
    Production  (7 usages found)
        IntelliJPlatformGradlePlugin.main  (7 usages found)
            IntelliJPlatformDependenciesHelper.kt  (7 usages found)
                IntelliJPlatformDependenciesHelper  (7 usages found)
                    collectDependencies  (2 usages found)
                        833 val version = "${plugin.pluginVersion}+${artifactPath.hash}"
                        861 val version = "$buildNumber+${platformPath.hash}"
                    createIntelliJPlatformBundledModule  (1 usage found)
                        795 val version = baseVersion.orElse(productInfo.map { it.version }).map { "$it+${platformPath.hash}" }.get()
                    createIntelliJPlatformBundledPlugin  (1 usage found)
                        763 val version = baseVersion.orElse(productInfo.map { it.version }).map { "$it+${artifactPath.hash}" }.get()
                    createIntelliJPlatformLocal  (1 usage found)
                        700 val version = "${localProductInfo.buildNumber}+${artifactPath.hash}"
                    createIntelliJPlatformLocalPlugin  (1 usage found)
                        897 val version = (plugin.pluginVersion ?: "0.0.0") + "+" + artifactPath.hash
                    createJetBrainsRuntimeLocal  (1 usage found)
                        939 val version = javaVendorVersion.removePrefix(name).trim('-') + "+" + artifactPath.hash

```

To understand the below it is important to understand how Gradle generates hashes for the transform cache. E.g. 
`/home/user/.gradle/caches/8.10.2/transforms/3eefd31b055e78c772b75249fc0f0276/transformed/ideaIU-242-EAP-SNAPSHOT/plugins/JavaEE` that transform hash `3eefd31b055e78c772b75249fc0f0276` is MD5 generated by https://github.com/gradle/gradle/blob/6236f2642795dc014e11eb5065c5e0d3e1e376bc/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformWorkspaceIdentity.java#L26 which may take into account many things but never absolute path:
- artifact name
- file hash (content)
- directory tree content hash
- Not really sure, but seems correct: artifact dependency coordinates or the way it was added to the dependencies, meaning that if we add a usual dependency on a lib it is one hash, but if we then add another with strict dependency version it will be another hash.

Docs https://docs.gradle.org/current/userguide/artifact_transforms.html#artifact_transforms_with_caching

This is bad for a number of reasons:
- I may be wrong, but it may be not following the convention https://maven.apache.org/guides/getting-started/index.html#what-is-a-snapshot-version It seems like you can add something before `o x.(y+1)-SNAPSHOT`  but not at the end.
- That absolute path usually points somewhere ilike `/home/user/.gradle/caches/8.10.2/transforms/3eefd31b055e78c772b75249fc0f0276/transformed/ideaIU-242-EAP-SNAPSHOT/plugins/JavaEE` which we hash and make a part of artifact coordinates. It can hurt things like https://docs.gradle.org/current/userguide/build_cache_debugging.html#caching_relocation_test I am not really sure if it breaks the build cache, but it does messes up transforms cache for sure, see below.
- Hashes will be different on different machines, which might make it difficult to commit some things into VCS is those dependencies get listed.
- With certain conditions met it breaks Gradle dependency locking https://docs.gradle.org/current/userguide/dependency_locking.html
- This one may be related maybe not, but https://docs.gradle.org/current/userguide/dependency_verification.html is not usable at all. It may be a separate issue, I will provide details as well. See https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779

### Relevant log output or stack trace

_No response_

### Steps to reproduce
See tests in https://github.com/JetBrains/intellij-platform-gradle-plugin/pull/1780
https://github.com/AlexanderBartash/intellij-platform-plugin-template/tree/dep-lock-bug

To reproduce the problem run:
`./gradlew --write-locks clean build`

this will generate a bunch of lock files, with strict rules on our all dependencies, including transitive.

`./gradlew build`

It will complain that a bunch (looks somewhat random, but probably is not) dependencies changed, because
now they have a different +number hash ending, which changed.

I think it happens, probably, because Gradle dependency locking works by simply adding to the project all those strict dependencies from the lock files to the project and lets dependency conflict resolution do the work
https://docs.gradle.org/current/userguide/dependency_constraints_conflicts.html#resolving_version_conflicts

If I am correct, then, it means, that we will probably have two platform dependencies, which may cause ExtractorTransformer to run again, since the version is different, creating new plugin dependencies, with new hashes, which makes dependency conflict resolution impossible. Or it may be that the presence of the dependency constraint causes the transform to run again, creating new artifact version (since the hash of the absolute path), which causes dependency locks to fail.

By default Gradle expects different behavior from transforms:
https://docs.gradle.org/current/userguide/artifact_transforms.html#executing_artifact_transforms
> can be run in parallel
> are cacheable
> are reusable (if separate resolutions used by different tasks require the same transform to be executed on the same artifacts, the transform results will be cached and shared)

I do not quite understand why some dependencies break that way and some don't.

```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache state could not be cached: field `provider` of `org.gradle.internal.serialize.codecs.core.ProviderBackedFileCollectionSpec` bean found in field `element` of `java.util.Collections$SingletonList` bean found in field `elements` of `org.gradle.internal.serialize.codecs.core.ResolutionBackedFileCollectionSpec` bean found in field `__classpathSnapshot__` of `org.jetbrains.kotlin.gradle.tasks.KotlinCompile$ClasspathSnapshotProperties` bean found in field `__classpathSnapshotProperties__` of task `:compileKotlin` of type `org.jetbrains.kotlin.gradle.tasks.KotlinCompile`: error writing value of type 'org.gradle.api.internal.provider.DefaultProvider'
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not resolve bundledPlugin:com.intellij.database:2023.3.8+994.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:com.intellij.database' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:2023.3.8+994'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:{strictly 2023.3.8+947}' because of the following reason: dependency was locked to version '2023.3.8+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.javaee:2023.3.8+703' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.persistence:2023.3.8+420' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.hibernate:233.15619.7+77' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.javaee.jpa:233.15619.7+982' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:Tomcat:2023.3.8+267' (default) --> 'bundledPlugin:com.intellij.javaee.app.servers.integration:233.15619.7+879' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'

   > Could not resolve bundledPlugin:com.intellij.diagram:2023.3.8+670.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:com.intellij.diagram' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.diagram:2023.3.8+670'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.diagram:{strictly 2023.3.8+247}' because of the following reason: dependency was locked to version '2023.3.8+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:2023.3.8+994' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+670'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.persistence:2023.3.8+420' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:2023.3.8+994' (default) --> 'bundledPlugin:JavaScript:233.15619.7+835' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.javaee.jpa:233.15619.7+982' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.microservices.jvm:233.15619.7+433' (default) --> 'bundledPlugin:com.jetbrains.restClient:233.15619.7+532' (default) --> 'bundledPlugin:com.intellij.microservices.ui:233.15619.7+324' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'

   > Could not resolve bundledPlugin:Tomcat:2023.3.8+267.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:Tomcat' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:Tomcat:2023.3.8+267'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:Tomcat:{strictly 2023.3.8+84}' because of the following reason: dependency was locked to version '2023.3.8+84'

   > Could not resolve bundledPlugin:org.jetbrains.idea.maven:2023.3.8+493.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:org.jetbrains.idea.maven' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:org.jetbrains.idea.maven:2023.3.8+493'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:org.jetbrains.idea.maven:{strictly 2023.3.8+222}' because of the following reason: dependency was locked to version '2023.3.8+222'

> There are 7 more failures with identical causes.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
Configuration cache entry discarded due to serialization error.
```

### Gradle IntelliJ Plugin version

2.1.0

### Gradle version

8.10.2

### Operating System

Linux

### Link to build, i.e. failing GitHub Action job

_No response_

I was able to confirm that ExtractorTransformer runs twice when dependency locks are enabled, generated & actually used.

What is described below was done without my fix: https://github.com/JetBrains/intellij-platform-gradle-plugin/pull/1780

So I did the next experiment in https://github.com/AlexanderBartash/intellij-platform-plugin-template/tree/dep-lock-bug
1. I killed Gradle daemons and removed `.gradle` in home.
2. Then: `./gradlew build`
[build-ExtractorTransformer-0.log](https://github.com/user-attachments/files/17214223/build-ExtractorTransformer-0.log)
ExtractorTransformer run for the first time and created
`/home/sasha/.gradle/caches/8.10.2/transforms/2f4561d46bc638866dfa4bcb9a564517/transformed/ideaIU-2023.3.8`
```
Selected primary task 'build' from project :
Resource missing. [HTTP GET: https://repo.maven.apache.org/maven2/com/jetbrains/intellij/idea/ideaIU/2023.3.8/ideaIU-2023.3.8.pom]
Downloading https://cache-redirector.jetbrains.com/www.jetbrains.com/intellij-repository/releases/com/jetbrains/intellij/idea/ideaIU/2023.3.8/ideaIU-2023.3.8.pom to /home/sasha/.gradle/.tmp/gradle_download4070131720423878460bin
Downloading https://cache-redirector.jetbrains.com/www.jetbrains.com/intellij-repository/releases/com/jetbrains/intellij/idea/ideaIU/2023.3.8/ideaIU-2023.3.8.zip to /home/sasha/.gradle/.tmp/gradle_download15370965056771427860bin
Transforming ideaIU-2023.3.8.zip (com.jetbrains.intellij.idea:ideaIU:2023.3.8) with ExtractorTransformer
Caching disabled for ExtractorTransformer: /home/sasha/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIU/2023.3.8/7cb6ca200b78b921b30a70b5057b952b5bfebd2a/ideaIU-2023.3.8.zip because:
  Caching not enabled.
[org.jetbrains.intellij.platform] Extracting archive '/home/sasha/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIU/2023.3.8/7cb6ca200b78b921b30a70b5057b952b5bfebd2a/ideaIU-2023.3.8.zip' to directory '/home/sasha/.gradle/caches/8.10.2/transforms/2f4561d46bc638866dfa4bcb9a564517-9020d4ce-416c-4c55-8c12-e0e7d791771b/transformed/ideaIU-2023.3.8'.
[org.jetbrains.intellij.platform] Resolving the content directory in '/home/sasha/.gradle/caches/8.10.2/transforms/2f4561d46bc638866dfa4bcb9a564517-9020d4ce-416c-4c55-8c12-e0e7d791771b/transformed/ideaIU-2023.3.8'.
[org.jetbrains.intellij.platform] The content directory is '/home/sasha/.gradle/caches/8.10.2/transforms/2f4561d46bc638866dfa4bcb9a564517-9020d4ce-416c-4c55-8c12-e0e7d791771b/transformed/ideaIU-2023.3.8'.
[org.jetbrains.intellij.platform] Extracting to '/home/sasha/.gradle/caches/8.10.2/transforms/2f4561d46bc638866dfa4bcb9a564517-9020d4ce-416c-4c55-8c12-e0e7d791771b/transformed/ideaIU-2023.3.8' completed.
```
3. Then I generated locks with: `./gradlew --write-locks build`
[build-ExtractorTransformer-1.log](https://github.com/user-attachments/files/17214187/build-ExtractorTransformer-1.log)
Transform cache was reused, as expected.
4. Then I tried to build again: `./gradlew build`
I expect this build to succeed because we just generated locks and they should be valid. But ExtractorTransformer run again and created another `/home/sasha/.gradle/caches/8.10.2/transforms/47266ca8ccefa5fcf2e6b5b95ae5e09e/transformed/ideaIU-2023.3.8`. Which due to absolute path hashing changes dependencies version string and fails the build, because versions for the same artifacts from the previous step have different hashes, because there the cache was reused.
[build-ExtractorTransformer-2.log](https://github.com/user-attachments/files/17214188/build-ExtractorTransformer-2.log)
```log
Task name matched 'build'
Selected primary task 'build' from project :
Loaded state for lock with ID 'jetbrainsRuntime'
Loaded state for lock with ID 'intellijPlatformDependency'
Loaded state for lock with ID 'intellijPlatformDependency'
Transforming ideaIU-2023.3.8.zip (com.jetbrains.intellij.idea:ideaIU:2023.3.8) with ExtractorTransformer
Caching disabled for ExtractorTransformer: /home/sasha/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIU/2023.3.8/7cb6ca200b78b921b30a70b5057b952b5bfebd2a/ideaIU-2023.3.8.zip because:
  Caching not enabled.
[org.jetbrains.intellij.platform] Extracting archive '/home/sasha/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIU/2023.3.8/7cb6ca200b78b921b30a70b5057b952b5bfebd2a/ideaIU-2023.3.8.zip' to directory '/home/sasha/.gradle/caches/8.10.2/transforms/47266ca8ccefa5fcf2e6b5b95ae5e09e-9e355738-303e-4811-82d3-c11e9a4485b7/transformed/ideaIU-2023.3.8'.
[org.jetbrains.intellij.platform] Resolving the content directory in '/home/sasha/.gradle/caches/8.10.2/transforms/47266ca8ccefa5fcf2e6b5b95ae5e09e-9e355738-303e-4811-82d3-c11e9a4485b7/transformed/ideaIU-2023.3.8'.
[org.jetbrains.intellij.platform] The content directory is '/home/sasha/.gradle/caches/8.10.2/transforms/47266ca8ccefa5fcf2e6b5b95ae5e09e-9e355738-303e-4811-82d3-c11e9a4485b7/transformed/ideaIU-2023.3.8'.
[org.jetbrains.intellij.platform] Extracting to '/home/sasha/.gradle/caches/8.10.2/transforms/47266ca8ccefa5fcf2e6b5b95ae5e09e-9e355738-303e-4811-82d3-c11e9a4485b7/transformed/ideaIU-2023.3.8' completed.
```
```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache state could not be cached: field `provider` of `org.gradle.internal.serialize.codecs.core.ProviderBackedFileCollectionSpec` bean found in field `element` of `java.util.Collections$SingletonList` bean found in field `elements` of `org.gradle.internal.serialize.codecs.core.ResolutionBackedFileCollectionSpec` bean found in field `__classpathSnapshot__` of `org.jetbrains.kotlin.gradle.tasks.KotlinCompile$ClasspathSnapshotProperties` bean found in field `__classpathSnapshotProperties__` of task `:compileKotlin` of type `org.jetbrains.kotlin.gradle.tasks.KotlinCompile`: error writing value of type 'org.gradle.api.internal.provider.DefaultProvider'
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not resolve bundledPlugin:com.intellij.database:2023.3.8+994.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:com.intellij.database' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:2023.3.8+994'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:{strictly 2023.3.8+947}' because of the following reason: dependency was locked to version '2023.3.8+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.javaee:2023.3.8+703' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.persistence:2023.3.8+420' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.hibernate:233.15619.7+77' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.javaee.jpa:233.15619.7+982' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:Tomcat:2023.3.8+267' (default) --> 'bundledPlugin:com.intellij.javaee.app.servers.integration:233.15619.7+879' (default) --> 'bundledPlugin:com.intellij.database:233.15619.7+947'

   > Could not resolve bundledPlugin:com.intellij.diagram:2023.3.8+670.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:com.intellij.diagram' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.diagram:2023.3.8+670'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.diagram:{strictly 2023.3.8+247}' because of the following reason: dependency was locked to version '2023.3.8+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:2023.3.8+994' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+670'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.persistence:2023.3.8+420' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.database:2023.3.8+994' (default) --> 'bundledPlugin:JavaScript:233.15619.7+835' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.javaee.jpa:233.15619.7+982' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:com.intellij.spring:2023.3.8+706' (default) --> 'bundledPlugin:com.intellij.microservices.jvm:233.15619.7+433' (default) --> 'bundledPlugin:com.jetbrains.restClient:233.15619.7+532' (default) --> 'bundledPlugin:com.intellij.microservices.ui:233.15619.7+324' (default) --> 'bundledPlugin:com.intellij.diagram:233.15619.7+247'

   > Could not resolve bundledPlugin:Tomcat:2023.3.8+267.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:Tomcat' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:Tomcat:2023.3.8+267'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:Tomcat:{strictly 2023.3.8+84}' because of the following reason: dependency was locked to version '2023.3.8+84'

   > Could not resolve bundledPlugin:org.jetbrains.idea.maven:2023.3.8+493.
     Required by:
         root project :
      > Cannot find a version of 'bundledPlugin:org.jetbrains.idea.maven' that satisfies the version constraints:
           Dependency path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:org.jetbrains.idea.maven:2023.3.8+493'
           Constraint path 'org.jetbrains.plugins.template:IntelliJ Platform Plugin Template:2.0.1' --> 'bundledPlugin:org.jetbrains.idea.maven:{strictly 2023.3.8+222}' because of the following reason: dependency was locked to version '2023.3.8+222'

> There are 7 more failures with identical causes.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 32s
Configuration cache entry discarded due to serialization error.
```

I tried it with the fix https://github.com/JetBrains/intellij-platform-gradle-plugin/pull/1780 as well and ExtractorTransformer runs twice as well, the only difference is that now there are no hashes in the versions and the last build was successful.
Logs:
[pr1780-build-ExtractorTransformer-0.log](https://github.com/user-attachments/files/17214402/pr1780-build-ExtractorTransformer-0.log)
[pr1780-build-ExtractorTransformer-1.log](https://github.com/user-attachments/files/17214403/pr1780-build-ExtractorTransformer-1.log)
[pr1780-build-ExtractorTransformer-2.log](https://github.com/user-attachments/files/17214404/pr1780-build-ExtractorTransformer-2.log)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,7 @@
 # Changelog
 
+## [next]
+
 ### Fixed
 
 - Fixed issue #1778 by removing a hash of the absolute artifact path appended to the end of the version string. That hash made artifact version different on different PCs and also breaks Gradle dependency locking.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,10 @@
 # Changelog
 
-## [next]
+### Fixed
+
+- Fixed issue #1778 by removing a hash of the absolute artifact path appended to the end of the version string. That hash made artifact version different on different PCs and also breaks Gradle dependency locking.
+
+## [2.1.0]
 
 ### Added
 

--- a/api/IntelliJPlatformGradlePlugin.api
+++ b/api/IntelliJPlatformGradlePlugin.api
@@ -176,6 +176,7 @@ public final class org/jetbrains/intellij/platform/gradle/Constants$Sandbox {
 public final class org/jetbrains/intellij/platform/gradle/Constants$Tasks {
 	public static final field BUILD_PLUGIN Ljava/lang/String;
 	public static final field BUILD_SEARCHABLE_OPTIONS Ljava/lang/String;
+	public static final field CLEAN Ljava/lang/String;
 	public static final field COMPOSED_JAR Ljava/lang/String;
 	public static final field GENERATE_MANIFEST Ljava/lang/String;
 	public static final field INITIALIZE_INTELLIJ_PLATFORM_PLUGIN Ljava/lang/String;

--- a/api/IntelliJPlatformGradlePlugin.api
+++ b/api/IntelliJPlatformGradlePlugin.api
@@ -176,7 +176,6 @@ public final class org/jetbrains/intellij/platform/gradle/Constants$Sandbox {
 public final class org/jetbrains/intellij/platform/gradle/Constants$Tasks {
 	public static final field BUILD_PLUGIN Ljava/lang/String;
 	public static final field BUILD_SEARCHABLE_OPTIONS Ljava/lang/String;
-	public static final field CLEAN Ljava/lang/String;
 	public static final field COMPOSED_JAR Ljava/lang/String;
 	public static final field GENERATE_MANIFEST Ljava/lang/String;
 	public static final field INITIALIZE_INTELLIJ_PLATFORM_PLUGIN Ljava/lang/String;
@@ -205,6 +204,7 @@ public final class org/jetbrains/intellij/platform/gradle/Constants$Tasks {
 }
 
 public final class org/jetbrains/intellij/platform/gradle/Constants$Tasks$External {
+	public static final field CLEAN Ljava/lang/String;
 	public static final field COMPILE_JAVA Ljava/lang/String;
 	public static final field COMPILE_KOTLIN Ljava/lang/String;
 	public static final field INSTANCE Lorg/jetbrains/intellij/platform/gradle/Constants$Tasks$External;

--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingAndVerificationIntegrationTest.kt
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingAndVerificationIntegrationTest.kt
@@ -1,0 +1,108 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.intellij.platform.gradle
+
+import kotlin.io.path.fileSize
+import kotlin.io.path.readText
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+/**
+ * Tests dependency locking & verification in combination.
+ * [DependencyLockingIntegrationTest] + [DependencyVerificationIgnoreIntellijIntegrationTest] but no artifacts are
+ * ignored in this test.
+ * [Support dependency verification](https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779)
+ */
+class DependencyLockingAndVerificationIntegrationTest : IntelliJPlatformIntegrationTestBase(
+    resourceName = "dependency-locking-and-verification",
+) {
+
+    @Test
+    @Ignore("https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779")
+    fun `build plugin with dependency locks & hash verification`() {
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties,
+            args = listOf("--info", "--write-locks", "--write-verification-metadata", "md5,sha1,sha256,sha512")
+        ) {
+            assertLockFilesExist()
+            assertMetadataSignaturesGenerated()
+        }
+
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties
+        ) {
+            buildDirectory containsFile "libs/test-1.0.0.jar"
+            assertLockFilesExist()
+            assertMetadataSignaturesGenerated()
+        }
+    }
+
+    @Test
+    @Ignore("https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779")
+    fun `build plugin with dependency locks, hash & signature verification`() {
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties,
+            args = listOf("--info", "--write-locks", "--write-verification-metadata", "md5,sha1,sha256,sha512,pgp")
+        ) {
+            assertLockFilesExist()
+            assertMetadataSignaturesGenerated()
+            assertMetadataTrustedKeysGenerated()
+        }
+
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties
+        ) {
+            buildDirectory containsFile "libs/test-1.0.0.jar"
+            assertLockFilesExist()
+            assertMetadataSignaturesGenerated()
+            assertMetadataTrustedKeysGenerated()
+        }
+    }
+
+    private fun assertLockFilesExist() {
+        buildDirectory.resolve("../gradle/locks/root/gradle.lockfile").let {
+            assertExists(it)
+            assertTrue(0 < it.fileSize())
+        }
+        buildDirectory.resolve("../gradle/locks/root/gradle-buildscript.lockfile").let {
+            assertExists(it)
+            assertTrue(0 < it.fileSize())
+        }
+        buildDirectory.resolve("../gradle/locks/root/settings-gradle-buildscript.lockfile").let {
+            assertExists(it)
+            assertTrue(0 < it.fileSize())
+        }
+    }
+
+    private fun assertMetadataSignaturesGenerated() {
+        buildDirectory.resolve("../gradle/verification-metadata.xml").let {
+            assertExists(it)
+            val xmlText = it.readText(Charsets.UTF_8)
+            assertTrue(xmlText.contains("<components>"))
+            assertTrue(xmlText.contains("<component"))
+            assertTrue(xmlText.contains("<artifact"))
+            assertTrue(xmlText.contains("<md5"))
+            assertTrue(xmlText.contains("<sha1"))
+            assertTrue(xmlText.contains("<sha256"))
+            assertTrue(xmlText.contains("<sha512"))
+        }
+    }
+
+    private fun assertMetadataTrustedKeysGenerated() {
+        buildDirectory.resolve("../gradle/verification-metadata.xml").let {
+            assertExists(it)
+            val xmlText = it.readText(Charsets.UTF_8)
+            assertTrue(xmlText.contains("<trusted-keys>"))
+            assertTrue(xmlText.contains("<trusted-key"))
+        }
+    }
+}

--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingAndVerificationIntegrationTest.kt
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingAndVerificationIntegrationTest.kt
@@ -2,6 +2,7 @@
 
 package org.jetbrains.intellij.platform.gradle
 
+import org.jetbrains.intellij.platform.gradle.Constants.Tasks
 import kotlin.io.path.fileSize
 import kotlin.io.path.readText
 import kotlin.test.Ignore
@@ -22,8 +23,8 @@ class DependencyLockingAndVerificationIntegrationTest : IntelliJPlatformIntegrat
     @Ignore("https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779")
     fun `build plugin with dependency locks & hash verification`() {
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties,
             args = listOf("--info", "--write-locks", "--write-verification-metadata", "md5,sha1,sha256,sha512")
         ) {
@@ -32,8 +33,8 @@ class DependencyLockingAndVerificationIntegrationTest : IntelliJPlatformIntegrat
         }
 
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties
         ) {
             buildDirectory containsFile "libs/test-1.0.0.jar"
@@ -46,8 +47,8 @@ class DependencyLockingAndVerificationIntegrationTest : IntelliJPlatformIntegrat
     @Ignore("https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779")
     fun `build plugin with dependency locks, hash & signature verification`() {
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties,
             args = listOf("--info", "--write-locks", "--write-verification-metadata", "md5,sha1,sha256,sha512,pgp")
         ) {
@@ -57,8 +58,8 @@ class DependencyLockingAndVerificationIntegrationTest : IntelliJPlatformIntegrat
         }
 
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties
         ) {
             buildDirectory containsFile "libs/test-1.0.0.jar"

--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingIntegrationTest.kt
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingIntegrationTest.kt
@@ -1,0 +1,51 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.intellij.platform.gradle
+
+import kotlin.io.path.fileSize
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+/**
+ * [Support for dependency locking, do not use absolute paths](https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1778)
+ */
+class DependencyLockingIntegrationTest : IntelliJPlatformIntegrationTestBase(
+    resourceName = "dependency-locking",
+) {
+
+    @Test
+    fun `build plugin with dependency locks`() {
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties,
+            args = listOf("--info", "--write-locks")
+        ) {
+            assertLockFilesExist()
+        }
+
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties
+        ) {
+            buildDirectory containsFile "libs/test-1.0.0.jar"
+            assertLockFilesExist()
+        }
+    }
+
+    private fun assertLockFilesExist() {
+        buildDirectory.resolve("../gradle/locks/root/gradle.lockfile").let {
+            assertExists(it)
+            assertTrue(0 < it.fileSize())
+        }
+        buildDirectory.resolve("../gradle/locks/root/gradle-buildscript.lockfile").let {
+            assertExists(it)
+            assertTrue(0 < it.fileSize())
+        }
+        buildDirectory.resolve("../gradle/locks/root/settings-gradle-buildscript.lockfile").let {
+            assertExists(it)
+            assertTrue(0 < it.fileSize())
+        }
+    }
+}

--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingIntegrationTest.kt
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyLockingIntegrationTest.kt
@@ -2,6 +2,7 @@
 
 package org.jetbrains.intellij.platform.gradle
 
+import org.jetbrains.intellij.platform.gradle.Constants.Tasks
 import kotlin.io.path.fileSize
 import kotlin.test.Test
 import kotlin.test.assertTrue
@@ -16,8 +17,8 @@ class DependencyLockingIntegrationTest : IntelliJPlatformIntegrationTestBase(
     @Test
     fun `build plugin with dependency locks`() {
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties,
             args = listOf("--info", "--write-locks")
         ) {
@@ -25,8 +26,8 @@ class DependencyLockingIntegrationTest : IntelliJPlatformIntegrationTestBase(
         }
 
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties
         ) {
             buildDirectory containsFile "libs/test-1.0.0.jar"

--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyVerificationIgnoreIntellijIntegrationTest.kt
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyVerificationIgnoreIntellijIntegrationTest.kt
@@ -1,0 +1,91 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.intellij.platform.gradle
+
+import kotlin.io.path.readText
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+/**
+ * For this test `src/integrationTest/resources/dependency-verification-ignore-intellij/gradle/verification-metadata.xml`
+ * adds "bundledModule" & "bundledPlugin" to trusted artifacts, so that we can test dependency locking & verification
+ * in isolation from Intellij Artifacts.
+ *
+ * [Support for dependency locking, do not use absolute paths](https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1778)
+ * [Support dependency verification](https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779)
+ * @see DependencyLockingIntegrationTest
+ * @see DependencyLockingAndVerificationIntegrationTest
+ */
+class DependencyVerificationIgnoreIntellijIntegrationTest : IntelliJPlatformIntegrationTestBase(
+    resourceName = "dependency-verification-ignore-intellij",
+) {
+
+    @Test
+    fun `build plugin with dependency locks & hash verification ignoring intellij artifacts`() {
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties,
+            args = listOf("--info", "--write-verification-metadata", "md5,sha1,sha256,sha512")
+        ) {
+            assertMetadataSignaturesGenerated()
+        }
+
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties
+        ) {
+            buildDirectory containsFile "libs/test-1.0.0.jar"
+            assertMetadataSignaturesGenerated()
+        }
+    }
+
+    @Test
+    @Ignore("https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779")
+    fun `build plugin with dependency locks, hash & signature verification ignoring intellij artifacts`() {
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties,
+            args = listOf("--info", "--write-verification-metadata", "md5,sha1,sha256,sha512,pgp")
+        ) {
+            assertMetadataSignaturesGenerated()
+            assertMetadataTrustedKeysGenerated()
+        }
+
+        build(
+            Constants.Tasks.CLEAN,
+            Constants.Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties
+        ) {
+            buildDirectory containsFile "libs/test-1.0.0.jar"
+            assertMetadataSignaturesGenerated()
+            assertMetadataTrustedKeysGenerated()
+        }
+    }
+
+    private fun assertMetadataSignaturesGenerated() {
+        buildDirectory.resolve("../gradle/verification-metadata.xml").let {
+            assertExists(it)
+            val xmlText = it.readText(Charsets.UTF_8)
+            assertTrue(xmlText.contains("<components>"))
+            assertTrue(xmlText.contains("<component"))
+            assertTrue(xmlText.contains("<artifact"))
+            assertTrue(xmlText.contains("<md5"))
+            assertTrue(xmlText.contains("<sha1"))
+            assertTrue(xmlText.contains("<sha256"))
+            assertTrue(xmlText.contains("<sha512"))
+        }
+    }
+
+    private fun assertMetadataTrustedKeysGenerated() {
+        buildDirectory.resolve("../gradle/verification-metadata.xml").let {
+            assertExists(it)
+            val xmlText = it.readText(Charsets.UTF_8)
+            assertTrue(xmlText.contains("<trusted-keys>"))
+            assertTrue(xmlText.contains("<trusted-key"))
+        }
+    }
+}

--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyVerificationIgnoreIntellijIntegrationTest.kt
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/DependencyVerificationIgnoreIntellijIntegrationTest.kt
@@ -2,6 +2,7 @@
 
 package org.jetbrains.intellij.platform.gradle
 
+import org.jetbrains.intellij.platform.gradle.Constants.Tasks
 import kotlin.io.path.readText
 import kotlin.test.Ignore
 import kotlin.test.Test
@@ -24,8 +25,8 @@ class DependencyVerificationIgnoreIntellijIntegrationTest : IntelliJPlatformInte
     @Test
     fun `build plugin with dependency locks & hash verification ignoring intellij artifacts`() {
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties,
             args = listOf("--info", "--write-verification-metadata", "md5,sha1,sha256,sha512")
         ) {
@@ -33,8 +34,8 @@ class DependencyVerificationIgnoreIntellijIntegrationTest : IntelliJPlatformInte
         }
 
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties
         ) {
             buildDirectory containsFile "libs/test-1.0.0.jar"
@@ -46,8 +47,8 @@ class DependencyVerificationIgnoreIntellijIntegrationTest : IntelliJPlatformInte
     @Ignore("https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779")
     fun `build plugin with dependency locks, hash & signature verification ignoring intellij artifacts`() {
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties,
             args = listOf("--info", "--write-verification-metadata", "md5,sha1,sha256,sha512,pgp")
         ) {
@@ -56,8 +57,8 @@ class DependencyVerificationIgnoreIntellijIntegrationTest : IntelliJPlatformInte
         }
 
         build(
-            Constants.Tasks.CLEAN,
-            Constants.Tasks.BUILD_PLUGIN,
+            Tasks.External.CLEAN,
+            Tasks.BUILD_PLUGIN,
             projectProperties = defaultProjectProperties
         ) {
             buildDirectory containsFile "libs/test-1.0.0.jar"

--- a/src/integrationTest/resources/dependency-locking-and-verification/build.gradle.kts
+++ b/src/integrationTest/resources/dependency-locking-and-verification/build.gradle.kts
@@ -1,0 +1,128 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+import org.jetbrains.intellij.platform.gradle.TestFrameworkType
+
+val intellijPlatformTypeProperty = providers.gradleProperty("intellijPlatform.type")
+val intellijPlatformVersionProperty = providers.gradleProperty("intellijPlatform.version")
+
+version = "1.0.0"
+
+plugins {
+    id("org.jetbrains.kotlin.jvm")
+    id("org.jetbrains.intellij.platform")
+}
+
+kotlin {
+    jvmToolchain(17)
+}
+
+buildscript {
+    // https://docs.gradle.org/current/userguide/dependency_locking.html
+    dependencyLocking {
+        lockAllConfigurations()
+        lockFile = file("gradle/locks/root/gradle-buildscript.lockfile")
+        lockMode.set(LockMode.DEFAULT)
+        //ignoredDependencies.add()
+    }
+}
+// https://docs.gradle.org/current/userguide/dependency_locking.html
+dependencyLocking {
+    lockAllConfigurations()
+    // There seems to be no way to customize the location of settings-gradle.lockfile
+    lockFile = file("gradle/locks/root/gradle.lockfile")
+    lockMode.set(LockMode.DEFAULT)
+}
+
+repositories {
+    mavenCentral()
+
+    intellijPlatform {
+        defaultRepositories()
+    }
+}
+
+dependencies {
+    intellijPlatform {
+        create(intellijPlatformTypeProperty, intellijPlatformVersionProperty)
+        instrumentationTools()
+        testFramework(TestFrameworkType.Platform)
+
+        // This is important for bug reproduction because we need some dependencies in the test
+        // Use ./gradlew printBundledPlugins
+        bundledPlugins(
+            "AntSupport",
+            "ByteCodeViewer",
+            "Coverage",
+            "DevKit",
+            "Git4Idea",
+            "JUnit",
+            "Lombook Plugin",
+            "PerforceDirectPlugin",
+            "Subversion",
+            "TestNG-J",
+            "XPathView",
+            "XSLT-Debugger",
+            "com.android.tools.design",
+            "com.android.tools.idea.smali",
+            "com.intellij.completion.ml.ranking",
+            "com.intellij.configurationScript",
+            "com.intellij.copyright",
+            "com.intellij.dev",
+            "com.intellij.gradle",
+            "com.intellij.java",
+            "com.intellij.java-i18n",
+            "com.intellij.java.ide",
+            "com.intellij.marketplace",
+            "com.intellij.platform.images",
+            "com.intellij.plugins.eclipsekeymap",
+            "com.intellij.plugins.netbeanskeymap",
+            "com.intellij.plugins.visualstudiokeymap",
+            "com.intellij.properties",
+            "com.intellij.searcheverywhere.ml",
+            "com.intellij.settingsSync",
+            "com.intellij.tasks",
+            "com.intellij.tracing.ide",
+            "com.intellij.uiDesigner",
+            "com.jetbrains.codeWithMe",
+            "com.jetbrains.packagesearch.intellij-plugin",
+            "com.jetbrains.projector.libs",
+            "com.jetbrains.sh",
+            "com.jetbrains.space",
+            "hg4idea",
+            "intellij.indexing.shared.core",
+            "intellij.webp",
+            "org.editorconfig.editorconfigjetbrains",
+            "org.intellij.groovy",
+            "org.intellij.intelliLang",
+            "org.intellij.plugins.markdown",
+            "org.intellij.qodana",
+            "org.jetbrains.android",
+            "org.jetbrains.debugger.streams",
+            "org.jetbrains.idea.eclipse",
+            "org.jetbrains.idea.gradle.dsl",
+            "org.jetbrains.idea.maven",
+            "org.jetbrains.idea.maven.model",
+            "org.jetbrains.idea.maven.server.api",
+            "org.jetbrains.idea.reposearch",
+            "org.jetbrains.java.decompiler",
+            "org.jetbrains.kotlin",
+            "org.jetbrains.plugins.github",
+            "org.jetbrains.plugins.gradle",
+            "org.jetbrains.plugins.gradle.dependency.updater",
+            "org.jetbrains.plugins.gradle.maven",
+            "org.jetbrains.plugins.javaFX",
+            "org.jetbrains.plugins.terminal",
+            "org.jetbrains.plugins.textmate",
+            "org.jetbrains.plugins.yaml",
+            "org.toml.lang",
+            "tanvd.grazi",
+            "training"
+        )
+    }
+}
+
+intellijPlatform {
+    buildSearchableOptions = false
+    instrumentCode = false
+}
+
+

--- a/src/integrationTest/resources/dependency-locking-and-verification/gradle.properties
+++ b/src/integrationTest/resources/dependency-locking-and-verification/gradle.properties
@@ -1,0 +1,2 @@
+# Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+kotlin.stdlib.default.dependency=false

--- a/src/integrationTest/resources/dependency-locking-and-verification/gradle/verification-metadata.xml
+++ b/src/integrationTest/resources/dependency-locking-and-verification/gradle/verification-metadata.xml
@@ -1,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+<!-- https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779 -->
+<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
+   <configuration>
+      <verify-metadata>true</verify-metadata>
+      <verify-signatures>false</verify-signatures>
+      <keyring-format>armored</keyring-format>
+      <key-servers>
+         <key-server uri="https://keyserver.ubuntu.com"/>
+      </key-servers>
+   </configuration>
+</verification-metadata>

--- a/src/integrationTest/resources/dependency-locking-and-verification/settings.gradle.kts
+++ b/src/integrationTest/resources/dependency-locking-and-verification/settings.gradle.kts
@@ -1,0 +1,12 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+rootProject.name = "test"
+
+buildscript {
+    // https://docs.gradle.org/current/userguide/dependency_locking.html
+    dependencyLocking {
+        lockAllConfigurations()
+        lockFile = file("gradle/locks/root/settings-gradle-buildscript.lockfile")
+        lockMode.set(LockMode.DEFAULT)
+        //ignoredDependencies.add()
+    }
+}

--- a/src/integrationTest/resources/dependency-locking/build.gradle.kts
+++ b/src/integrationTest/resources/dependency-locking/build.gradle.kts
@@ -1,0 +1,130 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+import org.jetbrains.intellij.platform.gradle.TestFrameworkType
+
+val intellijPlatformTypeProperty = providers.gradleProperty("intellijPlatform.type")
+val intellijPlatformVersionProperty = providers.gradleProperty("intellijPlatform.version")
+
+version = "1.0.0"
+
+plugins {
+    id("org.jetbrains.kotlin.jvm")
+    id("org.jetbrains.intellij.platform")
+}
+
+kotlin {
+    jvmToolchain(17)
+}
+
+buildscript {
+    // https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1778
+    // https://docs.gradle.org/current/userguide/dependency_locking.html
+    dependencyLocking {
+        lockAllConfigurations()
+        lockFile = file("gradle/locks/root/gradle-buildscript.lockfile")
+        lockMode.set(LockMode.DEFAULT)
+        //ignoredDependencies.add()
+    }
+}
+// https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1778
+// https://docs.gradle.org/current/userguide/dependency_locking.html
+dependencyLocking {
+    lockAllConfigurations()
+    // There seems to be no way to customize the location of settings-gradle.lockfile
+    lockFile = file("gradle/locks/root/gradle.lockfile")
+    lockMode.set(LockMode.DEFAULT)
+}
+
+repositories {
+    mavenCentral()
+
+    intellijPlatform {
+        defaultRepositories()
+    }
+}
+
+dependencies {
+    intellijPlatform {
+        create(intellijPlatformTypeProperty, intellijPlatformVersionProperty)
+        instrumentationTools()
+        testFramework(TestFrameworkType.Platform)
+
+        // This is important for bug reproduction because we need some dependencies in the test
+        // Use ./gradlew printBundledPlugins
+        bundledPlugins(
+            "AntSupport",
+            "ByteCodeViewer",
+            "Coverage",
+            "DevKit",
+            "Git4Idea",
+            "JUnit",
+            "Lombook Plugin",
+            "PerforceDirectPlugin",
+            "Subversion",
+            "TestNG-J",
+            "XPathView",
+            "XSLT-Debugger",
+            "com.android.tools.design",
+            "com.android.tools.idea.smali",
+            "com.intellij.completion.ml.ranking",
+            "com.intellij.configurationScript",
+            "com.intellij.copyright",
+            "com.intellij.dev",
+            "com.intellij.gradle",
+            "com.intellij.java",
+            "com.intellij.java-i18n",
+            "com.intellij.java.ide",
+            "com.intellij.marketplace",
+            "com.intellij.platform.images",
+            "com.intellij.plugins.eclipsekeymap",
+            "com.intellij.plugins.netbeanskeymap",
+            "com.intellij.plugins.visualstudiokeymap",
+            "com.intellij.properties",
+            "com.intellij.searcheverywhere.ml",
+            "com.intellij.settingsSync",
+            "com.intellij.tasks",
+            "com.intellij.tracing.ide",
+            "com.intellij.uiDesigner",
+            "com.jetbrains.codeWithMe",
+            "com.jetbrains.packagesearch.intellij-plugin",
+            "com.jetbrains.projector.libs",
+            "com.jetbrains.sh",
+            "com.jetbrains.space",
+            "hg4idea",
+            "intellij.indexing.shared.core",
+            "intellij.webp",
+            "org.editorconfig.editorconfigjetbrains",
+            "org.intellij.groovy",
+            "org.intellij.intelliLang",
+            "org.intellij.plugins.markdown",
+            "org.intellij.qodana",
+            "org.jetbrains.android",
+            "org.jetbrains.debugger.streams",
+            "org.jetbrains.idea.eclipse",
+            "org.jetbrains.idea.gradle.dsl",
+            "org.jetbrains.idea.maven",
+            "org.jetbrains.idea.maven.model",
+            "org.jetbrains.idea.maven.server.api",
+            "org.jetbrains.idea.reposearch",
+            "org.jetbrains.java.decompiler",
+            "org.jetbrains.kotlin",
+            "org.jetbrains.plugins.github",
+            "org.jetbrains.plugins.gradle",
+            "org.jetbrains.plugins.gradle.dependency.updater",
+            "org.jetbrains.plugins.gradle.maven",
+            "org.jetbrains.plugins.javaFX",
+            "org.jetbrains.plugins.terminal",
+            "org.jetbrains.plugins.textmate",
+            "org.jetbrains.plugins.yaml",
+            "org.toml.lang",
+            "tanvd.grazi",
+            "training"
+        )
+    }
+}
+
+intellijPlatform {
+    buildSearchableOptions = false
+    instrumentCode = false
+}
+
+

--- a/src/integrationTest/resources/dependency-locking/gradle.properties
+++ b/src/integrationTest/resources/dependency-locking/gradle.properties
@@ -1,0 +1,2 @@
+# Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+kotlin.stdlib.default.dependency=false

--- a/src/integrationTest/resources/dependency-locking/settings.gradle.kts
+++ b/src/integrationTest/resources/dependency-locking/settings.gradle.kts
@@ -1,0 +1,13 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+rootProject.name = "test"
+
+buildscript {
+    // https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1778
+    // https://docs.gradle.org/current/userguide/dependency_locking.html
+    dependencyLocking {
+        lockAllConfigurations()
+        lockFile = file("gradle/locks/root/settings-gradle-buildscript.lockfile")
+        lockMode.set(LockMode.DEFAULT)
+        //ignoredDependencies.add()
+    }
+}

--- a/src/integrationTest/resources/dependency-verification-ignore-intellij/build.gradle.kts
+++ b/src/integrationTest/resources/dependency-verification-ignore-intellij/build.gradle.kts
@@ -1,0 +1,111 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+import org.jetbrains.intellij.platform.gradle.TestFrameworkType
+
+val intellijPlatformTypeProperty = providers.gradleProperty("intellijPlatform.type")
+val intellijPlatformVersionProperty = providers.gradleProperty("intellijPlatform.version")
+
+version = "1.0.0"
+
+plugins {
+    id("org.jetbrains.kotlin.jvm")
+    id("org.jetbrains.intellij.platform")
+}
+
+kotlin {
+    jvmToolchain(17)
+}
+
+repositories {
+    mavenCentral()
+
+    intellijPlatform {
+        defaultRepositories()
+    }
+}
+
+dependencies {
+    intellijPlatform {
+        create(intellijPlatformTypeProperty, intellijPlatformVersionProperty)
+        instrumentationTools()
+        testFramework(TestFrameworkType.Platform)
+
+        // This is important for bug reproduction because we need some dependencies in the test
+        // Use ./gradlew printBundledPlugins
+        bundledPlugins(
+            "AntSupport",
+            "ByteCodeViewer",
+            "Coverage",
+            "DevKit",
+            "Git4Idea",
+            "JUnit",
+            "Lombook Plugin",
+            "PerforceDirectPlugin",
+            "Subversion",
+            "TestNG-J",
+            "XPathView",
+            "XSLT-Debugger",
+            "com.android.tools.design",
+            "com.android.tools.idea.smali",
+            "com.intellij.completion.ml.ranking",
+            "com.intellij.configurationScript",
+            "com.intellij.copyright",
+            "com.intellij.dev",
+            "com.intellij.gradle",
+            "com.intellij.java",
+            "com.intellij.java-i18n",
+            "com.intellij.java.ide",
+            "com.intellij.marketplace",
+            "com.intellij.platform.images",
+            "com.intellij.plugins.eclipsekeymap",
+            "com.intellij.plugins.netbeanskeymap",
+            "com.intellij.plugins.visualstudiokeymap",
+            "com.intellij.properties",
+            "com.intellij.searcheverywhere.ml",
+            "com.intellij.settingsSync",
+            "com.intellij.tasks",
+            "com.intellij.tracing.ide",
+            "com.intellij.uiDesigner",
+            "com.jetbrains.codeWithMe",
+            "com.jetbrains.packagesearch.intellij-plugin",
+            "com.jetbrains.projector.libs",
+            "com.jetbrains.sh",
+            "com.jetbrains.space",
+            "hg4idea",
+            "intellij.indexing.shared.core",
+            "intellij.webp",
+            "org.editorconfig.editorconfigjetbrains",
+            "org.intellij.groovy",
+            "org.intellij.intelliLang",
+            "org.intellij.plugins.markdown",
+            "org.intellij.qodana",
+            "org.jetbrains.android",
+            "org.jetbrains.debugger.streams",
+            "org.jetbrains.idea.eclipse",
+            "org.jetbrains.idea.gradle.dsl",
+            "org.jetbrains.idea.maven",
+            "org.jetbrains.idea.maven.model",
+            "org.jetbrains.idea.maven.server.api",
+            "org.jetbrains.idea.reposearch",
+            "org.jetbrains.java.decompiler",
+            "org.jetbrains.kotlin",
+            "org.jetbrains.plugins.github",
+            "org.jetbrains.plugins.gradle",
+            "org.jetbrains.plugins.gradle.dependency.updater",
+            "org.jetbrains.plugins.gradle.maven",
+            "org.jetbrains.plugins.javaFX",
+            "org.jetbrains.plugins.terminal",
+            "org.jetbrains.plugins.textmate",
+            "org.jetbrains.plugins.yaml",
+            "org.toml.lang",
+            "tanvd.grazi",
+            "training"
+            )
+    }
+}
+
+intellijPlatform {
+    buildSearchableOptions = false
+    instrumentCode = false
+}
+
+

--- a/src/integrationTest/resources/dependency-verification-ignore-intellij/gradle.properties
+++ b/src/integrationTest/resources/dependency-verification-ignore-intellij/gradle.properties
@@ -1,0 +1,2 @@
+# Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+kotlin.stdlib.default.dependency=false

--- a/src/integrationTest/resources/dependency-verification-ignore-intellij/gradle/verification-metadata.xml
+++ b/src/integrationTest/resources/dependency-verification-ignore-intellij/gradle/verification-metadata.xml
@@ -1,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+<!-- https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1779 -->
+<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
+   <configuration>
+      <verify-metadata>true</verify-metadata>
+      <verify-signatures>false</verify-signatures>
+      <keyring-format>armored</keyring-format>
+      <key-servers>
+         <key-server uri="https://keyserver.ubuntu.com"/>
+      </key-servers>
+      <trusted-artifacts>
+         <trust group="bundledModule" reason="Testing"/>
+         <trust group="bundledPlugin" reason="Testing"/>
+      </trusted-artifacts>
+   </configuration>
+</verification-metadata>

--- a/src/integrationTest/resources/dependency-verification-ignore-intellij/settings.gradle.kts
+++ b/src/integrationTest/resources/dependency-verification-ignore-intellij/settings.gradle.kts
@@ -1,0 +1,2 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+rootProject.name = "test"

--- a/src/main/kotlin/org/jetbrains/intellij/platform/gradle/Constants.kt
+++ b/src/main/kotlin/org/jetbrains/intellij/platform/gradle/Constants.kt
@@ -130,7 +130,6 @@ object Constants {
     }
 
     object Tasks {
-        const val CLEAN = "clean"
         const val BUILD_PLUGIN = "buildPlugin"
         const val BUILD_SEARCHABLE_OPTIONS = "buildSearchableOptions"
         const val COMPOSED_JAR = "composedJar"
@@ -159,6 +158,7 @@ object Constants {
         const val VERIFY_PLUGIN_SIGNATURE = "verifyPluginSignature"
 
         object External {
+            const val CLEAN = "clean"
             const val COMPILE_JAVA = JavaPlugin.COMPILE_JAVA_TASK_NAME
             const val COMPILE_KOTLIN = "compileKotlin"
             const val JAR = JavaPlugin.JAR_TASK_NAME

--- a/src/main/kotlin/org/jetbrains/intellij/platform/gradle/Constants.kt
+++ b/src/main/kotlin/org/jetbrains/intellij/platform/gradle/Constants.kt
@@ -130,6 +130,7 @@ object Constants {
     }
 
     object Tasks {
+        const val CLEAN = "clean"
         const val BUILD_PLUGIN = "buildPlugin"
         const val BUILD_SEARCHABLE_OPTIONS = "buildSearchableOptions"
         const val COMPOSED_JAR = "composedJar"

--- a/src/main/kotlin/org/jetbrains/intellij/platform/gradle/artifacts/transform/CollectorTransformer.kt
+++ b/src/main/kotlin/org/jetbrains/intellij/platform/gradle/artifacts/transform/CollectorTransformer.kt
@@ -27,7 +27,6 @@ import org.jetbrains.intellij.platform.gradle.utils.asPath
 import org.jetbrains.intellij.platform.gradle.utils.platformPath
 import org.jetbrains.intellij.platform.gradle.utils.safelyCreatePlugin
 import java.nio.file.Path
-import kotlin.Throws
 import kotlin.io.path.createTempDirectory
 import kotlin.io.path.exists
 import kotlin.io.path.listDirectoryEntries

--- a/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt
+++ b/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt
@@ -40,9 +40,7 @@ import java.io.File
 import java.io.FileReader
 import java.nio.file.Path
 import java.util.*
-import kotlin.Throws
 import kotlin.io.path.*
-import kotlin.math.absoluteValue
 
 /**
  * Helper class for managing dependencies on the IntelliJ Platform in Gradle projects.
@@ -697,7 +695,7 @@ class IntelliJPlatformDependenciesHelper(
         localProductInfo.validateSupportedVersion()
 
         val type = localProductInfo.productCode.toIntelliJPlatformType()
-        val version = "${localProductInfo.buildNumber}+${artifactPath.hash}"
+        val version = localProductInfo.buildNumber
 
         writeIvyModule(Dependencies.LOCAL_IDE_GROUP, type.code, version) {
             IvyModule(
@@ -760,7 +758,7 @@ class IntelliJPlatformDependenciesHelper(
         }
 
         val artifactPath = requireNotNull(plugin.originalFile)
-        val version = baseVersion.orElse(productInfo.map { it.version }).map { "$it+${artifactPath.hash}" }.get()
+        val version = baseVersion.orElse(productInfo.map { it.version }).get()
 
         writeIvyModule(Dependencies.BUNDLED_PLUGIN_GROUP, id, version) {
             IvyModule(
@@ -792,7 +790,7 @@ class IntelliJPlatformDependenciesHelper(
             .let { requireNotNull(it) { "Specified bundledModule '$id' doesn't exist." } }
         val platformPath = platformPath.get()
         val artifactPaths = bundledModule.classPath.map { path -> platformPath.resolve(path).toIvyArtifact() }
-        val version = baseVersion.orElse(productInfo.map { it.version }).map { "$it+${platformPath.hash}" }.get()
+        val version = baseVersion.orElse(productInfo.map { it.version }).get()
 
         writeIvyModule(Dependencies.BUNDLED_MODULE_GROUP, id, version) {
             IvyModule(
@@ -830,7 +828,7 @@ class IntelliJPlatformDependenciesHelper(
                 val artifactPath = requireNotNull(plugin.originalFile)
                 val group = Dependencies.BUNDLED_PLUGIN_GROUP
                 val name = requireNotNull(plugin.pluginId)
-                val version = "${plugin.pluginVersion}+${artifactPath.hash}"
+                val version = "${plugin.pluginVersion}"
 
                 writeIvyModule(group, name, version) {
                     IvyModule(
@@ -858,7 +856,7 @@ class IntelliJPlatformDependenciesHelper(
                 val artifactPaths = it.classPath.map { path -> platformPath.resolve(path).toIvyArtifact() }
                 val group = Dependencies.BUNDLED_MODULE_GROUP
                 val name = it.name
-                val version = "$buildNumber+${platformPath.hash}"
+                val version = buildNumber
 
                 writeIvyModule(group, name, version) {
                     IvyModule(
@@ -894,7 +892,7 @@ class IntelliJPlatformDependenciesHelper(
             pluginManager.safelyCreatePlugin(pluginPath).getOrThrow()
         }
 
-        val version = (plugin.pluginVersion ?: "0.0.0") + "+" + artifactPath.hash
+        val version = (plugin.pluginVersion ?: "0.0.0")
         val name = plugin.pluginId ?: artifactPath.name
 
         writeIvyModule(Dependencies.LOCAL_PLUGIN_GROUP, name, version) {
@@ -936,7 +934,7 @@ class IntelliJPlatformDependenciesHelper(
         requireNotNull(javaVersion)
 
         val name = javaVendorVersion.substringBefore(javaVersion).trim('-')
-        val version = javaVendorVersion.removePrefix(name).trim('-') + "+" + artifactPath.hash
+        val version = javaVendorVersion.removePrefix(name).trim('-')
 
         writeIvyModule(Dependencies.LOCAL_JETBRAINS_RUNTIME_GROUP, name, version) {
             IvyModule(
@@ -1093,9 +1091,6 @@ class IntelliJPlatformDependenciesHelper(
         version = version,
         extension = "zip",
     )
-
-    private val Path.hash
-        get() = (absolutePathString().hashCode().absoluteValue % 1000).toString()
 
     internal fun buildJetBrainsRuntimeVersion(
         version: String,

--- a/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt
+++ b/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt
@@ -40,6 +40,7 @@ import java.io.File
 import java.io.FileReader
 import java.nio.file.Path
 import java.util.*
+import kotlin.Throws
 import kotlin.io.path.*
 
 /**
@@ -133,6 +134,7 @@ class IntelliJPlatformDependenciesHelper(
             .listDirectoryEntries()
             .filter { it.isDirectory() }
             .mapNotNull { path ->
+                // TODO: try not to parse all plugins at once
                 pluginManager.safelyCreatePlugin(path)
                     .onFailure { log.warn(it.message.orEmpty()) }
                     .getOrNull()
@@ -828,7 +830,7 @@ class IntelliJPlatformDependenciesHelper(
                 val artifactPath = requireNotNull(plugin.originalFile)
                 val group = Dependencies.BUNDLED_PLUGIN_GROUP
                 val name = requireNotNull(plugin.pluginId)
-                val version = "${plugin.pluginVersion}"
+                val version = requireNotNull(plugin.pluginVersion)
 
                 writeIvyModule(group, name, version) {
                     IvyModule(
@@ -892,7 +894,7 @@ class IntelliJPlatformDependenciesHelper(
             pluginManager.safelyCreatePlugin(pluginPath).getOrThrow()
         }
 
-        val version = (plugin.pluginVersion ?: "0.0.0")
+        val version = plugin.pluginVersion ?: "0.0.0"
         val name = plugin.pluginId ?: artifactPath.name
 
         writeIvyModule(Dependencies.LOCAL_PLUGIN_GROUP, name, version) {

